doc/source/dask-on-ray.rst

```diff
@@ -25,50 +25,73 @@ Here's an example:
+   # each compute call.
+   dask.config.set(scheduler=ray_dask_get)
+
+   df = dd.from_pandas(pd.DataFrame(np.random.randint(0, 100, size=(1024, 2)), columns=["age", "grade"]))
+   print(df.groupby(["age"]).mean().compute())
 
-   x = inc(1)
-   y = inc(2)
-   z = add(x, y)
-   # The Dask scheduler submits the underlying task graph to Ray.
-   z.compute(scheduler=ray_dask_get)
 
 Why use Dask on Ray?
 
-   1. If you'd like to create data analyses using the familiar NumPy and Pandas
-      APIs provided by Dask and execute them on a production-ready distributed
-      task execution system like Ray.
-   2. If you'd like to use Dask and Ray libraries in the same application
-      without having two different task execution backends.
+   1. If you'd like to create data analyses using the familiar NumPy and Pandas APIs provided by Dask and execute them on a fast, fault-tolerant distributed task execution system geared towards production, like Ray.
+   2. If you'd like to use Dask and Ray libraries in the same application without having two different clusters.
    3. To take advantage of Ray-specific features such as the
       :ref:`cluster launcher <ref-automatic-cluster>` and
       :ref:`shared-memory store <memory>`.
 
 Note that for execution on a Ray cluster, you should *not* use the
 `Dask.distributed <https://distributed.dask.org/en/latest/quickstart.html>`__
 client; simply use plain Dask and its collections, and pass ``ray_dask_get``
-to ``.compute()`` calls. Follow the instructions for
+to ``.compute()`` calls or set the scheduler in one of the other ways detailed `here <https://docs.dask.org/en/latest/scheduling.html#configuration>`__. Follow the instructions for
 :ref:`using Ray on a cluster <using-ray-on-a-cluster>` to modify the
 ``ray.init()`` call.
 
 Dask-on-Ray is an ongoing project and is not expected to achieve the same performance as using Ray directly.
 
+================================================
+Custom optimization for Dask DataFrame shuffling
+================================================
+
+We have also created a custom Dask DataFrame optimizer that leverages Ray's ability to
+execute multiple-return tasks in order to speed up shuffling by as much as 10x on Ray.
+Simply set the `dataframe_optimize` configuration option to our optimizer function, similar to how you specify the Dask-on-Ray scheduler:
+
+.. code-block:: python
+
+   import ray
+   from ray.util.dask import ray_dask_get, dataframe_optimize
+   import dask.dataframe as dd
+   import numpy as np
+   import pandas as pd
+   import time
+
+   # Start Ray.
+   # Tip: If you're connecting to an existing cluster, use ray.init(address="auto").
+   ray.init()
+
+   # Set the scheduler to ray_dask_get, and set the Dask DataFrame optimizer to our
+   # custom optimization function, this time using the config setter as a context manager.
+   with dask.config.set(scheduler=ray_dask_get, dataframe_optimize=dataframe_optimize):
```

Comment:
Can we enable these by default on ray.init()?

Reply:
We can, although I think that would be less good for a few reasons:

1. Having the core `ray.init()` path reach out and set Dask configs seems like a poor separation of concerns, and may be odd for those using both Dask and Ray but _not_ trying to use Dask-_on_-Ray.
2. If the user wants to compare the two, they wouldn't want that config set globally, and might find that confusing.

I personally think that making the user explicitly set a config to enable Dask-on-Ray is the better way to go. If you think that this configuration is getting ugly (and could get uglier if we add more Ray-specific optimization passes), we could provide a helper like `enable_dask_on_ray()` that hides the complexity of manually setting these Dask configs.

Reply:
Makes sense. I think adding something like `ray.util.enable_dask_integration()` would be very helpful for users.

Reply:
Agreed.

Reply:
This should be pretty quick, right? @clarkzinzow, do you want to do that in this PR?

Reply:
That should be very quick, yes, but I'll keep this PR focused on docs and open another PR that adds that helper.